File: ProjectEuler/Problems/Problem029.cs
using System.Collections.Generic;

namespace ProjectEuler.Problems
{
    public class Problem029 : Problem<long>
    {
        protected override long CalculateAnswer()
        {
            // unique a^b terms where 2 <= a <= 100 and 2 <= b <= 100

            var minA = 2L;
            var minB = 2;
            var maxA = 100L;
            var maxB = 100;

            // generate a list of all (possibly repetitive) terms in prime factorization format
            var terms = new List<PrimeFactorization>();
            for (var a = minA; a <= maxA; ++a)
            {
                var factorizationOfA = MathHelpers.PrimeFactorization(a);
                for (var b = minB; b <= maxB; ++b)
                {
                    var factorization = factorizationOfA.Copy();
                    foreach (var entry in factorization.Entries)
                    {
                        entry.Power *= b;
                    }
                    terms.Add(factorization);
                }
            }

            // compare each pair of terms and remove any duplicates
            for (var i = 0; i < terms.Count; ++i)
            {
                var term1 = terms[i];
                var j = i + 1;
                while (j < terms.Count)
                {
                    var term2 = terms[j];
                    if (term1.Equals(term2)) terms.RemoveAt(j);
                    else ++j;
                }
            }

            return terms.Count;
        }
    }
}

File: ProjectEulerPython/problems/problem_029.py
from problems.problem import Problem

class Problem029(Problem):

  def calculate_answer(self) -> int:
    min_a = 2
    min_b = 2
    max_a = 100
    max_b = 100

    terms = {} # map only allows unique keys
    for a in range(min_a, max_a + 1):
      for b in range(min_b, max_b + 1):
        product = 1
        for i in range(0, b):
          product *= a
        terms[product] = True

    return len(terms)